### DISCUSSION TITLE
How will this handle extremely large amounts of data?

### DISCUSSION THREAD

**Discussion by ochompsky** [stashapp/stash#5585] [General]
I'm talking half a million files, over 200TB.  I know phash will take a good amount of time, but im also wondering how the UI/database will be able to handle it.  Anyone using this for very large libraries?

**Comment by DogmaDragon** [comment]
We have people with petabytes of storage using it without any issues. It might be slower than with a fresh instance, but it should be easily usable.

**Comment by stg-annon** [comment]
there are some caveats to larger libraries, you'll want to offload any Blobs to the filesystem to keep the DB from becoming massive, you'll likely want to not calculate MD5 for videos as it will need to read all of your data rather than select parts of it

I think these are default options now, but I'm not sure, I haven't started a new Stash from scratch in a while
